desc/objectives/utils.py

```diff
@@ -97,34 +97,61 @@ def factorize_linear_constraints(objective, constraint):  # noqa: C901
         A = A[:, cols]
     assert A.shape[1] == xp.size
 
-    # fixed just means there is a single element in A, so A_ij*x_j = b_i
-    fixed_rows = np.where(np.count_nonzero(A, axis=1) == 1)[0]
-    # indices of x that are fixed = cols of A where rows have 1 nonzero val.
-    _, fixed_idx = np.where(A[fixed_rows])
-    unfixed_rows = np.setdiff1d(np.arange(A.shape[0]), fixed_rows)
-    unfixed_idx = np.setdiff1d(np.arange(objective.dim_x), fixed_idx)
-    if len(fixed_rows):
-        # something like 0.5 x1 = 2 is the same as x1 = 4
-        b = put(b, fixed_rows, b[fixed_rows] / np.sum(A[fixed_rows], axis=1))
-        A = put(
-            A,
-            Index[fixed_rows, :],
-            A[fixed_rows] / np.sum(A[fixed_rows], axis=1)[:, None],
-        )
-        xp = put(xp, fixed_idx, b[fixed_rows])
-        # Some values might be fixed, but they still show up in other constraints
-        # this is where the fixed cols have >1 nonzero val.
-        # For fixed variables, we delete that row and col of A, but that means
-        # we need to subtract the fixed value from b so that the equation is balanced.
-        # e.g., 2 x1 + 3 x2 + 1 x3 = 4 ; 4 x1 = 2
-        # combining gives 3 x2 + 1 x3 = 3, with x1 now removed
-        b = put(
-            b,
-            unfixed_rows,
-            b[unfixed_rows] - A[unfixed_rows][:, fixed_idx] @ b[fixed_rows],
-        )
-    A = A[unfixed_rows][:, unfixed_idx]
-    b = b[unfixed_rows]
+    # will store the global index of the unfixed rows with shifted indices by 1
```

Comment:
why shift the indices by one?

Reply:
That is actually related to how I delete fixed indices later in the code. For some reason, I chose to set the fixed parameters to 0 and then delete the 0 values, so the initial array shouldn't have 0. At the time I wrote it, it felt convenient to use np.nonzero() to find nonzero elements. If it is hard to follow, I can change it to something else

Reply:
I used np.delete now